### PR TITLE
code

### PR DESCRIPTION
**Pull request by Tahirbhalli**
This is a Testing Repo, where I Write tests with JEST.
Tests for cisher cipher
Tests for Strings
Tests for calculator
Tests for Array
for Run:
> npm run test

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :clap: 
You have done a great job setting up the project and writing the initial tests. We however need to make the following changes before we can merge.
- [x] add modules for ceaser cipher and array analysis
- [x] add tests for the above modules. make sure each module has it's own test file. For example, ceaser.js would have a test file as ceaser.test.js.
- [x] the index.test.js file contains tests for calculator.js and stringoperation.js modules. you need to split the file into calculator.test.js for calculator module and stringoperation.test.js for the stringoperation module.
- [x] add a title to this PR and a detailed description of what features this PR covers.

### Martin Afani

File: src/js/stringoperation.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const Strings = (() => {
+  const capitalize = (str) => str.toUpperCase();
```
Comment (Oghenebrume50) [inline src/js/stringoperation.js]: Capitalize should make only the first character capital see

File: src/js/ceaser.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable no-extend-native */
+const alpabets = 'abcdefghijklmnopqrstuvwxyz';
+String.prototype.Ceaser = function (shift) {
```
Comment (Oghenebrume50) [inline src/js/ceaser.js:4]: The caser cipher function should take in the string and the shift as parameters, in order to test against multiple parameters this is the best way to implement it. 
Also you do not necessarily need to modify the String prototype, but if fine if you want to keep it that way :+1:

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

 Hi @Tahirbhalli good job working on this and kudos for making changes outlined by the previous reviewer :slightly_smiling_face:  please look into my comments below and try to make changes were appropriate :slightly_smiling_face:  

- Capitalize and  Caesar is currently failing and I have pointed out the reason below please make changes, modify your test, and submit for another review :slightly_smiling_face: 

Please submit for another review when you are done :slightly_smiling_face:  

**RAPHAEL NORIODE** :mask: stay_safe

File: src/js/Array.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  }
+  return max;
+};
+Array.prototype.avg = function () {
+  let sum = 0;
+  for (let index = 0; index < this.length; index += 1) {
+    sum += this[index];
+  }
+  return sum / this.length;
+};
```
Comment (shshamim63) [inline src/js/Array.js:27]: [optional]
You were supposed to create a function which takes an array as an input and returns an object like the following ->
Suppose the method name as arrayAnalysis and an array [1, 2, 3, 4, 5, 6]. So if you call arrayAnalysis([1, 2, 3, 4, 5, 6]) should return.
```js
{
    average: 3.50,
    min: 1,
    max: 6,
    length: 6,
}
```
I am really sorry this was not mentioned earlier.

**Review by shshamim63 — APPROVED** [review]
## :1234: Approved
___
Hi @Tahirbhalli :wave: 
The tests are working fine. all of them are separated in different files, great job :clap: . But the function related to the array you were supposed to build is not covering the instruction. I am sorry that it was not mentioned earlier. So I am adding it as optional. Please check the comment below.
___
#### This :+1: is from me, you can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Pull request merged**